solr/core/src/java/org/apache/solr/handler/IndexFetcher.java

```diff
@@ -271,7 +274,34 @@ public IndexFetcher(@SuppressWarnings({"rawtypes"})final NamedList initArgs, fin
     String httpBasicAuthPassword = (String) initArgs.get(HttpClientUtil.PROP_BASIC_AUTH_PASS);
     myHttpClient = createHttpClient(solrCore, httpBasicAuthUser, httpBasicAuthPassword, useExternalCompression);
   }
-  
+
+  private void setLeaderUrl(String leaderUrl) {
+    if (leaderUrl != null) {
+      ShardHandlerFactory shardHandlerFactory = solrCore.getCoreContainer().getShardHandlerFactory();
+      if (shardHandlerFactory instanceof HttpShardHandlerFactory) {
+        ZkController zkController = solrCore.getCoreContainer().getZkController();
+        ClusterState clusterState = zkController == null ? null : zkController.getClusterState();
+        try {
+          ((HttpShardHandlerFactory) shardHandlerFactory).getWhitelistHostChecker()
+              .checkWhitelist(clusterState, null, Collections.singletonList(leaderUrl));
```

Comment:
When using Solr with only leader/follower replication and not with SolrCloud, this whitelist check is failing - unless `-Dsolr.disable.shardsWhitelist=true` is passed. In my point of view it does not make sense to have to pass this flag when SolrCloud is not being used. What do you think?

Reply:
This is by design and on purpose, please see documentation of the feature here https://solr.apache.org/guide/8_8/distributed-requests.html#configuring-the-shardhandlerfactory

In leader/follower installs, it is expected that you configure the `shardsWhitelist` setting in solr.xml with a list of all your shards, to avoid user queries with shards parameter being able to connect to any host on the network.

Reply:
I'm not using distributed search, just old style master/slave or in new terms leader/follower replication. There is no configuration about shards, SolrCloud or whatsoever, but this whitelist check throws the following exception: `Invalid URL syntax in "shards" parameter: null`

I don't believe that this is the desired behaviour when configuring only a `solr.ReplicationHandler` in solrconfig.xml. In the documentation https://solr.apache.org/guide/8_8/index-replication.html there is no word mentioned about shards.

Reply:
Ok, this is for validation of `leaderUrl` in replication handler. In SOLR-15340 the setting is renamed to `allowUrls` which better reflects the generic intent for it, such as validating any user/config provided URL in the system.

As this sounds like a bug, please file a NEW Jira issue with a clear description on how to reproduce the issue, including the Exception stacktrace you see, so we can get a bug fix in the next bugfix release.

Reply:
Hmm, yes... I tried now tried twice to create a jira ticket and spent quite some time formatting it nicely. Both times (with different IPs) when submitting the ticket I got `Your request has been blocked. If you feel this is in error, please let us know at: abuse@infra.apache.org`. Srsly, wtf... I already sent an email there, but didn't get a response.